sycl/test-e2e/Adapters/sycl-ls-gpu-default-any.cpp

```diff
@@ -1,4 +1,5 @@
 // REQUIRES: gpu
+// REQUIRES: build-and-run-mode
```

Comment:
Not currently, I can make a PR for changes test-e2e readme adding those details.

For now here's the extra considerations I've come across with the split testing:
- In tests that build multiple binaries and execute them we need to make sure that each binary has a different name so that they are not overwritten, and we can transfer all the binaries to the run system.
- For XFAILS we have two scenarios to consider now. 
  - If the expected failure occurs during compilation we need to mark the XFAIL on a feature that is device-agnostic (i.e., `XFAIL: linux` or `XFAIL: *`). If in this situation we mark a test as XFAIL with a device-specific feature we will get a normal failure, since the `build-only` mode will not have any device-specific features.
  - If the expected failure occurs during run-time we will need to mark the test with XFAIL on a device specific feature (i.e., `XFAIL: gpu`, `XFAIL: level_zero`), or if its expected to always fail on run-time we can use `XFAIL: run-mode`. If in this situation we mark a test as `XFAIL` with a device-agnostic feature we will get an unexpected pass on the build stage, since the test will be able to compile.
- To do the test splitting we classify `RUN:` lines as either build or run. if a line contains `%{run}`, `%{run-unfiltered-devices}` or `%if run-mode` they are classified as run-lines, otherwise they are classified as build-lines. 
  - Previously we could get away with having lines that executed the test binary without `%{run}` or `%{run-unfiltered-devices}`, however now this will usually create a failure at the build stage as we will try to execute the binary without the appropriate devices.
  - The vast majority of `RUN:` lines that do not use `%{run}`/`%{run-unfiltered-devices}` are needed to compile the binary, or need to be ran prior to the execution (but not necessarily on the machine doing the execution). in the case that we need to run a command after executing the test binaries, and using either of the run expansions does not make sense, we can use `%if run-mode` to make sure it runs on the running system instead of the build system.
- Right now the build-only mode does not support logic to properly assess the features in REQUIRES/UNSUPPORTED to know if a test should be built in the environment (or for `spir64`), Only tests that are marked `REQUIRES: build-and-run-mode` or `UNSUPPORTED: true` are skipped. so if a test will fail building for the build environment we have on CI or for spir64 we will need to mark this as `REQUIRES: build-and-run-mode`. This is only temporary, and I am working on a way to not need these types of requires statements.
- Other than that there are the AOT tests, and tests which do not compile anything at all. Which I mentioned in my response to Alexey above.

Reply:
That's a good chunk of info :+1: to documenting this in the README.

Reply:
@kbenzie @AlexeySachkov Made #16409 to add info in the README